src/test/java/de/tum/in/www1/artemis/localvcci/LocalCIServiceTest.java

```diff
@@ -47,14 +57,55 @@ class LocalCIServiceTest extends AbstractSpringIntegrationLocalCILocalVCTest {
     @Autowired
     private AeolusTemplateService aeolusTemplateService;
 
+    @Autowired
+    private SharedQueueProcessingService sharedQueueProcessingService;
+
+    @Autowired
+    private HazelcastInstance hazelcastInstance;
+
+    protected IQueue<LocalCIBuildJobQueueItem> queuedJobs;
+
+    protected IMap<Long, LocalCIBuildJobQueueItem> processingJobs;
+
     @Test
     @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
     void testReturnCorrectBuildStatus() {
         userUtilService.addUsers(TEST_PREFIX, 1, 0, 0, 1);
         Course course = programmingExerciseUtilService.addCourseWithOneProgrammingExercise();
         ProgrammingExercise exercise = exerciseUtilService.getFirstExerciseWithType(course, ProgrammingExercise.class);
         ProgrammingExerciseStudentParticipation participation = participationUtilService.addStudentParticipationForProgrammingExercise(exercise, TEST_PREFIX + "student1");
+
+        // remove listener to avoid triggering build job processing
+        sharedQueueProcessingService.removeListener();
+
+        JobTimingInfo jobTimingInfo = new JobTimingInfo(ZonedDateTime.now(), ZonedDateTime.now().plusMinutes(1), ZonedDateTime.now().plusMinutes(2));
+        BuildConfig buildConfig = new BuildConfig("echo 'test'", "test", "test", "test", null, null, false, false, false, null);
+        RepositoryInfo repositoryInfo = new RepositoryInfo("test", null, RepositoryType.USER, "test", "test", "test", null, null);
+
+        LocalCIBuildJobQueueItem job1 = new LocalCIBuildJobQueueItem("1", "job1", "address1", participation.getId(), course.getId(), 1, 1, 1,
+                de.tum.in.www1.artemis.domain.enumeration.BuildStatus.SUCCESSFUL, repositoryInfo, jobTimingInfo, buildConfig, null);
+        LocalCIBuildJobQueueItem job2 = new LocalCIBuildJobQueueItem("2", "job2", "address1", participation.getId(), course.getId(), 1, 1, 1,
+                de.tum.in.www1.artemis.domain.enumeration.BuildStatus.SUCCESSFUL, repositoryInfo, jobTimingInfo, buildConfig, null);
+
+        queuedJobs = hazelcastInstance.getQueue("buildJobQueue");
+        processingJobs = hazelcastInstance.getMap("processingJobs");
+
+        // No build jobs for the participation are queued or building
         assertThat(continuousIntegrationService.getBuildStatus(participation)).isEqualTo(BuildStatus.INACTIVE);
+
+        queuedJobs.add(job1);
+        processingJobs.put(1L, job2);
+
+        // At least one build job for the participation is queued
+        assertThat(continuousIntegrationService.getBuildStatus(participation)).isEqualTo(BuildStatus.QUEUED);
+        queuedJobs.clear();
+
+        // No build jobs for the participation are queued, but at least one is building
+        assertThat(continuousIntegrationService.getBuildStatus(participation)).isEqualTo(BuildStatus.BUILDING);
+        processingJobs.clear();
+
+        // init to activate queue listener again
+        sharedQueueProcessingService.init();
```

Comment:
Can you move this into the AfterEach method? This will not be executed if the test fails.